### PR TITLE
[frisby]: add `FrisbyResponse` type

### PR DESCRIPTION
**Pull request by EkeMinusYou**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/vlucas/frisby/blob/master/src/frisby/spec.js#L246>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/frisby/index.d.ts [hunk 1]
```diff
@@ -48,17 +48,29 @@ export class FrisbySpec {
     inspectStatus(): FrisbySpec;
     patch(url: string, params?: {}): FrisbySpec;
     post(url: string, params?: {}): FrisbySpec;
-    promise(): Promise<nodeFetch.Response>;
+    promise(): Promise<FrisbyResponse>;
```
Comment (EkeMinusYou) [inline types/frisby/index.d.ts:51]: Actually,  `promise()` function returns `FrisbyResponse`, not `nodeFetch.Response`. So I fixed.

https://github.com/vlucas/frisby/blob/master/src/frisby/spec.js#L303

File: types/frisby/index.d.ts [hunk 2]
```diff
@@ -48,17 +48,29 @@ export class FrisbySpec {
     inspectStatus(): FrisbySpec;
     patch(url: string, params?: {}): FrisbySpec;
     post(url: string, params?: {}): FrisbySpec;
-    promise(): Promise<nodeFetch.Response>;
+    promise(): Promise<TResult>;
```
Comment (EkeMinusYou) [inline types/frisby/index.d.ts:51]: Actually,  `promise()` function returns last response such as `FrisbyResponse`, not `nodeFetch.Response`. So I fixed.

https://github.com/vlucas/frisby/blob/master/src/frisby/spec.js#L303

File: types/frisby/tslint.json [hunk 3]
```diff
@@ -1,3 +1,6 @@
 {
-    "extends": "dtslint/dt.json"
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "strict-export-declare-modifiers": false
```
Comment (weswigham) [inline types/frisby/tslint.json:4]: please remove this lint exception, thanks~

**Review by weswigham — CHANGES_REQUESTED** [review]

File: types/frisby/tslint.json (hunk 3 above)
Comment (EkeMinusYou) [reply types/frisby/tslint.json]: Fixed. Thank you.

Also,  `FrisbyResponse` class is not directly referenced, so I changed it to `interface`.

**Review by weswigham — APPROVED** [review]

**Comment by EkeMinusYou** [comment]
Ready to merge

**Pull request merged**